opnode/l2/api.go

```diff
@@ -137,15 +140,54 @@ func (payload *ExecutionPayload) ParentHash() common.Hash {
 }
 
 func (payload *ExecutionPayload) Transactions() types.Transactions {
-	res := make([]*types.Transaction, len(payload.TransactionsField))
-	for i, t := range payload.TransactionsField {
-		res[i] = new(types.Transaction)
-		err := res[i].UnmarshalBinary(t)
-		if err != nil {
-			panic(err)
+	return payload.unmarshaledTxs
```

Comment:
As soon as we change the `TransactionsField` this gets out of sync. We use this type not just to read from L1, but also to *write* to L2. So it seems prone to bugs, and I don't particularly like the duplication of all JSON things either.